### DISCUSSION TITLE
Intersection between a box and a line

### DISCUSSION THREAD

**Discussion by IlariaFichera** [pyvista/pyvista#5368] [Q&A]
Hello,
I have created the following code with pyvista. This is just a trial before a more complicated code. The objective is to create a line passing through the box, create a box and get the intersection between the line and the box. In this particular case, the line passes through the boxing two points on two different box surfaces (box boundaries). I would like to get the length of the line within the two points on the two different box surfaces. The code created is the following:
`import pyvista as pv
import numpy as np

p1 = pv.Box(bounds=(0,3,0,3,0,3),level=10)
l1 = pv.Line(pointa=(-0.5, -0.5, 0), pointb=(4.0, 2.5, 2.5),resolution = 10)
intersection, p1_split, l1_split = p1.intersection(l1)

pl = pv.Plotter()
_ = pl.add_mesh(p1, style='wireframe')
_ = pl.add_mesh(l1, style='wireframe')
pl.show()`

It gives me the following error:
ERROR:root:No points to subdivide
ERROR:root:No points to subdivide

2023-12-20 12:42:37.273 (1629.764s) [                ]            vtkMath.cxx:797   WARN| vtkMath::Jacobi: Error extracting eigenfunctions
2023-12-20 12:42:37.274 (1629.765s) [                ]    vtkPointLocator.cxx:868    ERR| vtkPointLocator (00000281C5C43F60): No points to subdivide
2023-12-20 12:42:37.274 (1629.765s) [                ]vtkIntersectionPolyData:2410  WARN| No Intersection between objects 
2023-12-20 12:42:37.277 (1629.767s) [                ]            vtkMath.cxx:797   WARN| vtkMath::Jacobi: Error extracting eigenfunctions
2023-12-20 12:42:37.277 (1629.768s) [                ]    vtkPointLocator.cxx:868    ERR| vtkPointLocator (00000281C5C44560): No points to subdivide
2023-12-20 12:42:37.278 (1629.768s) [                ]vtkIntersectionPolyData:2410  WARN| No Intersection between objects 

What exactly does this mean? Can I create an intersection between a line and a box?
At the moment, the line is defined with some points, but in the future, the line will be defined by multiple points, therefore, can I get somehow the coordinates points of intersection with the two surfaces of the box?

Thank you very much

Regards,

Ilaria

**Comment by darikg** [comment]
The intersection filter is for calculating the intersection between two surfaces, I think you want some sort of ray trace, like this:

```python
p1 = pv.Box(bounds=(0,3,0,3,0,3), level=10)
pts, cell_idx = p1.ray_trace((-0.5, -0.5, 0), (4.0, 2.5, 2.5))
pts, cell_idx
```

> the line will be defined by multiple points

Meaning a series of line segments? I think you can either do a for-loop over the above, or check out [multi_ray_trace](https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.PolyDataFilters.multi_ray_trace.html) for something like

```python
l1 = pv.Line(pointa=(-0.5, -0.5, 0), pointb=(4.0, 2.5, 2.5),resolution = 10)
pts0 = l1.points[:-1]  # (10, 3) array of starting points
pts1 = l1.points[1:]  
direction = pts1 - pts0  #(10, 3) array of direction vectors
p1.triangulate().multi_ray_trace(pts0, direction)
```

multi_ray_trace has some extra dependencies, and isn't quite the same because it checks for intersections with the infinitely long ray instead of the finite-length line segment.
